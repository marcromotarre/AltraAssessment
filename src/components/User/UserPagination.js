import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const UserPaginationContainer = styled.button`
  margin: 0;
  border: 0;
  width: 100%;
  padding: 10px 30px 10px 10px;
  height: 40px;
  border-bottom: 1px solid #e3e3e3;
  background-color: #f8f8f8;
  display: flex;
  justify-content: center;
  align-items: center;
  label {
    font-family: 'HelveticaNeue-Bold';
    color: #707070;
    font-size: 16px;
  }
`;

const UserPagination = ({ text, onClick }) => {
  const onClickContainer = () => {
    onClick();
  };
  return (
    <UserPaginationContainer onClick={onClickContainer}>
      <label>{text}</label>
    </UserPaginationContainer>
  );
};

UserPagination.propTypes = {
  text: PropTypes.string,
  onClick: PropTypes.func,
};

UserPagination.defaultProps = {
  text: 'Load more users',
  onClick: () => {},
};

export default UserPagination;
