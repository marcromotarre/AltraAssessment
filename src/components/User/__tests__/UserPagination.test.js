import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/react';
import UserPagination from '../UserPagination';

/** this component should:
 *  - render button style like UserElement with a label with a text defined by props
 *          -  when click the button call props function
 */

const props = {
  onClick: jest.fn(),
  text: 'text',
};

describe('User Pagination Component tests', () => {
  it('render UserPagination Snapshot', () => {
    const tree = renderer.create(<UserPagination {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render default props text if no text passed on props', () => {
    const { container } = render(<UserPagination />);
    const label = container.querySelector('label');
    expect(label.innerHTML).toBe('Load more users');
  });

  it('should render props text', () => {
    const { container } = render(<UserPagination {...props} />);
    const label = container.querySelector('label');
    expect(label.innerHTML).toBe(props.text);
  });

  it('call props onClick function when on click', () => {
    const { container } = render(<UserPagination {...props} />);
    const button = container.querySelector('button');
    expect(props.onClick).toHaveBeenCalledTimes(0);
    fireEvent.click(button);
    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});
