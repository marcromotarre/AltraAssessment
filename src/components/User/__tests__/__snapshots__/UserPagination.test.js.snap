// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Pagination Component tests render UserPagination Snapshot 1`] = `
<button
  className="sc-AxjAm rteci"
  onClick={[Function]}
>
  <label>
    text
  </label>
</button>
`;
